### PR TITLE
Additions to the message module

### PR DESCRIPTION
**Pull request by neithernut**
This PR introduces some new interfaces to the `message` module of the library and refactors a few things on the way. In preparation for #67.

### CODE REVIEW

**Comment by neithernut** [comment]
I have yet to test whether these changes break anything.

**Comment by neithernut** [comment]
Tested superficially. Should work.

File: lib/src/error.rs [hunk 1]
```diff
@@ -40,6 +40,11 @@ error_chain! {
             display("Malformed head refernece: {}", name)
         }
 
+        TrailerFormatError {
```
Comment (matthiasbeyer) [inline lib/src/error.rs]: Passing in the trailer here would be nice, so we can display it!

Comment (neithernut) [inline lib/src/error.rs]: You mean the string which we tried to parse into a trailer.

Comment (matthiasbeyer) [inline lib/src/error.rs]: Yep, exactly.

**Pull request merged**